src/wallet/wallet.cpp

```diff
@@ -1073,43 +1070,41 @@ bool CWallet::WitnessBucketCommitment(uint256 &commitment,
             {
                 BOOST_FOREACH(const uint256 &bucket_commitment, pour.commitments)
                 {
-                    std::vector<bool> commitment_bv(ZC_CM_SIZE * 8);
-                    std::vector<bool> index;
-                    std::vector<unsigned char> commitment_value(bucket_commitment.begin(), bucket_commitment.end());
-                    libzerocash::convertBytesVectorToVector(commitment_value, commitment_bv);
-                    assert(tree.insertElement(commitment_bv, index));
-
-                    if (bucket_commitment == commitment) {
-                        // We've found it! Now, we construct a witness.
-                        res = true;
-                        commitment_index = index;
+                    if (witness) {
+                        witness->append(bucket_commitment);
+                    } else {
+                        tree.append(bucket_commitment);
+
+                        if (bucket_commitment == commitment) {
+                            witness = tree.witness();
+                        }
                     }
                 }
             }
         }
 
-        {
-            std::vector<unsigned char> newrt_v(32);
-            tree.getRootValue(newrt_v);
-            current_anchor = uint256(newrt_v);
+        if (witness) {
+            current_anchor = witness->root();
+        } else {
+            current_anchor = tree.root();
         }
 
         // Consistency check: we should be able to find the current tree
         // in our CCoins view.
-        libzerocash::IncrementalMerkleTree dummy_tree(INCREMENTAL_MERKLE_TREE_DEPTH);
+        ZCIncrementalMerkleTree dummy_tree;
         assert(pcoinsTip->GetAnchorAt(current_anchor, dummy_tree));
 
         pindex = chainActive.Next(pindex);
     }
 
-    if (res) {
-        assert(tree.getWitness(commitment_index, path));
-    }
+    if (witness) {
+        path = witness->path();
+        final_anchor = current_anchor;
 
-    path_index = libzerocash::convertVectorToInt(commitment_index);
-    final_anchor = current_anchor;
+        return true;
+    }
 
-    return res;
+    return false;
```

Comment:
This results in an undocumented change in the side-effects of `CWallet::WitnessBucketCommitment()`. But the only two callers (in `rpcwallet.cpp`) weren't and aren't using the values of `path` and `final_anchor` if it returned false, so this should be okay.